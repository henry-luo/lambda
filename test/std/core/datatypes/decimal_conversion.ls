// Test: Decimal Type Conversion
// Category: core/datatypes
// Type: positive
// Expected: 1

// String to decimal conversion
let from_string = "123.456" as decimal
from_string
let from_int = 42 as decimal
from_int
let from_float = 3.14 as decimal  
from_float
let price = 19.99m
price
let price_str = price as string  
price_str
let price_float = price as float  
price_float
let price_int = price as int     
price_int
let scientific = "1.23e-4" as decimal  
scientific
let with_commas = "1,234.56" as decimal  
with_commas
let large_decimal = "79228162514264337593543950335" as decimal  
large_decimal
let small_decimal = "0.0000000000000000000000000001" as decimal  
small_decimal
1
