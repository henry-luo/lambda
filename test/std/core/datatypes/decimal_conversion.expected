123.456
42
3.14
19.99
"19.99"
19.99
19
0.000123
1234.56
79228162514264337593543950335
0.0000000000000000000000000001
1
